src/coreclr/jit/lsrabuild.cpp

```diff
@@ -2321,32 +2321,31 @@ void LinearScan::buildIntervals()
     numPlacedArgLocals = 0;
     placedArgRegs      = RBM_NONE;
 
-    BasicBlock* predBlock = nullptr;
     BasicBlock* prevBlock = nullptr;
 
     // Initialize currentLiveVars to the empty set.  We will set it to the current
     // live-in at the entry to each block (this will include the incoming args on
     // the first block).
     VarSetOps::AssignNoCopy(compiler, currentLiveVars, VarSetOps::MakeEmpty(compiler));
+    const bool floatingPointUsed = compiler->compFloatingPointUsed;
```

Comment:
As I understand it `compFloatingPointUsed` is meant to be set before we get to LSRA. What is the case where it is not set and then gets set later? I think it would be better to have some `blockUsedFloatingPoint` variable that is deterministically set for each block and that LSRA can use for decisions that involve only the particular block.

Reply:
Ideally yes and we should audit the cases where that is not happening. I remember running into this a while back and had to rely on it only after `buildIntervals()` where it was guaranteed to set correctly.

Reply:
In `LinearScan::BuildDef`, we have a check that sets `compFloatingPointUsed` to true if the tree we're building the definition for can't use integer registers. I believe you're understanding of `compFloatingPointUsed` is correct -- LSRA doesn't seem to be using it all that judiciously, hence this PR. I'll try your suggestion in a follow-up.

Reply:
> In LinearScan::BuildDef, we have a check that sets compFloatingPointUsed to true if the tree we're building the definition for can't use integer registers.

We duplicate this check in `LinearScan::identifyCandidates`, so we have a few spots in LSRA to clean up.

Reply:
The problem with this is that it is going to come with TP regressions when someone decides to set `compFloatingPointUsed` correctly in the future. That's why I would like it to be explicit if we are changing LSRA to make "block-local" decisions in some cases where that is ok.

Reply:
It looks like the only place is when we create kills for calls. So likely we could just have some bool `needsToKillFloatRegs` that LSRA uses for this purpose, and that is block-local in MinOpts and global (using `compFloatingPointUsed`) in FullOpts.

Reply:
> The problem with this is that it is going to come with TP regressions when someone decides to set compFloatingPointUsed correctly in the future. That's why I would like it to be explicit if we are changing LSRA to make "block-local" decisions in some cases where that is ok.

That's valid, I'll open a follow-up PR for this.